lesson21/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage("Communication with the server is broken.",element);
+        return;
+    }
+    const json = await response.json();
+    return json.data;
+}
+
+const fetchContentsData = (endPointURL,element,ms) => new Promise(resolve => setTimeout(() => resolve(fetchData(endPointURL,element)),ms));
+
+const table = document.getElementById("js-table");
+
+const initUsersData = async () => {
+    loading.showLoading(table);
+    try{
+        const usersData = await fetchContentsData(endPointURL.usersData,table,500);
+        if (!usersData.length) {
+            createErrorMessage("No user.",table);
+            return;
+        }
+        renderTableElements(usersData);
+        sortTableBody(usersData);
+    }catch(error){
+        console.error(error);
+    }finally{
+       loading.removeLoading(table);
+    }
+}
+
+const renderTableElements = usersData => {
+    renderTableHeader();
+    renderTableBody(usersData);
+}
+
+const usersTableColumn = {
+    "id"     : "ID",
+    "name"   : "名前",
+    "gender" : "性別",
+    "age"    : "年齢"
+}
+
+const renderTableHeader = () => {
+    const thead = document.createElement("thead");
+    const tr = document.createElement("tr");
+    thead.className = "bg-slate-500";
+
+    for (const [userKey, userValue] of Object.entries(usersTableColumn)) {
+        const th = createAttributedElements({
+            tag:"th",
+            valuesByAttributes:{
+                class:"text-sm text-white px-6 py-4",
+            },
+            str:userValue
+        });
+
+        tr.appendChild(th);
+
+        addSortButtons(userKey,userValue,th);
+    };
+
+    table.appendChild(thead).appendChild(tr);
+}
+
+const renderTableBody = usersData => {
+    const tbody = document.createElement("tbody");
+    tbody.id = "js-usersTableBody";
+
+    for(const user of usersData){
+        const fragment = document.createDocumentFragment();
+        const tr = document.createElement("tr");
+
+        Object.keys(usersTableColumn).forEach((column) => {
+            const td = createAttributedElements({
+                tag:"td",
+                valuesByAttributes:{
+                    class:"border border-gray-300 px-4 py-2"
+                },
+                str:user[column]
+            })
+            fragment.appendChild(tr).appendChild(td);
+        });
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtons = [
+    {dataOrder:"asc",src:"../img/asc.svg", alt:"asc-image"},
+    {dataOrder:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {dataOrder:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortByUsersTableColumn = {
+    "id" : "ID"
+}
+
+const addSortButtons = (userKey,userValue,th) => {
+    for(const value of Object.values(sortByUsersTableColumn)){
+        userValue === value && createSortButtons(userKey,th) ;
+    }
+}
+
```

Comment:
IMO
I thought it would be better to append created here instead of passing `th` as an argument
```suggestion
const addSortButtons = (userKey,userValue,th) => {
    for(const value of Object.values(sortByUsersTableColumn)){
       userValue === value && th.appendChild(createSortButtons(userKey, th));
    }
}

```

Reply:
Certainly it seems that there is no need to pass it as an argument.
I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/dc583cd97d6ce22a6fd1dbaf63ff774d77808d6d